feat(discover): added add to discover button to saved discover queries

edwardgou-sentry opened the pull request:
Added `Add to Discover` button to saved discover queries:

![image](https://user-images.githubusercontent.com/83961295/130679508-99fc0dbe-ded9-46e7-a3a1-bc186b6d866c.png)
![image](https://user-images.githubusercontent.com/83961295/130679543-a166b446-36a6-442f-9b97-9a3321c7bb9d.png)

Keeping `feat/discover-add-dashboard-widget` as target branch for now until merged due to dependancy.

edwardgou-sentry committed:
feature flag

edwardgou-sentry committed:
progress

edwardgou-sentry committed:
progress

edwardgou-sentry committed:
progress

edwardgou-sentry committed:
progress

edwardgou-sentry committed:
validation

edwardgou-sentry committed:
new dashboard

edwardgou-sentry committed:
Save widget from discover

edwardgou-sentry committed:
added query menu in add widget to dashboard modal from discover

edwardgou-sentry committed:
create new dashboard and widget from discover

edwardgou-sentry committed:
Merge branch 'master' of github.com:getsentry/sentry into feat/discover-add-dashboard-widget

edwardgou-sentry committed:
undo hide query

edwardgou-sentry committed:
removed disabled prop

edwardgou-sentry committed:
removed todos

edwardgou-sentry committed:
Discover Query Menu

edwardgou-sentry committed:
button styling

edwardgou-sentry committed:
tests

edwardgou-sentry committed:
constructWidgetFromQuery refactor and tests

edwardgou-sentry committed:
tests

edwardgou-sentry committed:
tests

edwardgou-sentry committed:
Merge branch 'master' of github.com:getsentry/sentry into feat/discover-add-dashboard-widget

edwardgou-sentry committed:
Merge branch 'master' of github.com:getsentry/sentry into feat/discover-add-dashboard-widget

edwardgou-sentry committed:
added add to dashboard menu item to saved queries in discover saved query page

edwardgou-sentry committed:
discoverAddToDashboardModal accepts start, end, period for rendering
chart based on time selected

edwardgou-sentry committed:
Merge branch 'master' of github.com:getsentry/sentry into feat/discover-add-dashboard-widget

edwardgou-sentry committed:
fixed view bug when adding widget to default-overview dashboard for the
first time
use saved query title as default widget title
updated references to mapErrors and normalizeQueries to use util
implementations
also check for dashboards-edit feature

edwardgou-sentry committed:
updated tests

edwardgou-sentry committed:
tsc

edwardgou-sentry committed:
consolidated discoverAddToDashboardModal into addDashboardWidgetModal

edwardgou-sentry committed:
fixed typo

edwardgou-sentry committed:
refactored some stuff

edwardgou-sentry committed:
addDashboardWidgetModal

shruthilayaj commented on static/app/views/eventsV2/queryList.tsx:
```diff
@@ -208,6 +225,21 @@ class QueryList extends React.Component<Props> {
             <ContextMenu>
+              <Feature
+                organization={organization}
+                features={['connect-discover-and-dashboards']}
```
We should check for both `dashboard-edit` and `connect-discover-and-dashboards` features here

edwardgou-sentry replied:
Ahh nice catch, forgot to include some of the feedback changes from the other PR into this one! Updated!

shruthilayaj commented on static/app/views/eventsV2/queryList.tsx:
```diff
@@ -83,6 +85,21 @@ class QueryList extends React.Component<Props> {
+      const {organization} = this.props;
+      event.preventDefault();
+      event.stopPropagation();
+      openAddDashboardWidgetModal({
```
Can we pass defaultTitle here?

edwardgou-sentry replied:
👍

shruthilayaj commented on static/app/views/eventsV2/queryList.tsx:
```diff
@@ -208,6 +225,21 @@ class QueryList extends React.Component<Props> {
+              >
+                {({hasFeature}) =>
+                  hasFeature && (
+                    <MenuItem
```
the prebuilt queries should also have the option to add to dashboard (since you can add them from the query detail page)

edwardgou-sentry replied:
Updated! 👍

shruthilayaj commented on static/app/components/modals/addDashboardWidgetModal.tsx:
```diff
@@ -340,10 +350,18 @@ class AddDashboardWidgetModal extends React.Component<Props, State> {
     const state = this.state;
     const errors = state.errors;
 
+    const querySelection = statsPeriod
```
can we do this instead of casting global selection below?
```suggestion
    const querySelection: GlobalSelection = statsPeriod
```

shruthilayaj replied:
hmm in this case, should we also be persisting environment and projects selection to the dashboard? If it gets too complicated, I think we can just use the default selection as the first pass?

edwardgou-sentry replied:
Updated this! Had to make a slight tweak to how `querySelection` is constructed to please typescript but functionally it works the same.

edwardgou-sentry committed:
added dashboard-edit feature check
added defaultTitle
added context menu for prebuilt queries
updated GlobalSelection type to allow null period

edwardgou-sentry committed:
GlobalSelection stuff

wmak commented on static/app/utils/discover/eventView.tsx:
```diff
@@ -413,7 +413,7 @@ class EventView {
     });
 
-    return new EventView({
+    const eventView = new EventView({
```
nit: is this change needed?

edwardgou-sentry replied:
oops no this is not needed, left it in by accident during testing. cleaned it up!

wmak reviewed (approved):
LGTM

edwardgou-sentry committed:
clean up

edwardgou-sentry committed:
tsc feedback

edwardgou-sentry committed:
Merge branch 'master' of github.com:getsentry/sentry into feat/discover-add-dashboard-widget

edwardgou-sentry committed:
removed unneeded async

edwardgou-sentry committed:
fixed widget chart rerendering on empty field changes

edwardgou-sentry committed:
Merge branch 'master' of github.com:getsentry/sentry into feat/saved-queries-add-to-dashboard
